src/token/token.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
+import { Message } from 'discord.js';
+import { TokenCacheService } from './token-cache.service';
+
+@Injectable()
+export class TokenService {
+  constructor(private readonly tokenCacheService: TokenCacheService) {}
+
+  public async createToken(message: Message) {
+    const serverId = message.guild.id;
+    try {
+      const token = await this.tokenCacheService.getNewToken(serverId, [
+        'Data.Read',
```

Comment:
Should this be hardcoded?

Reply:
As there are only two scopes its okay I think. When making the Scoeps more granular we have to make them adjustable from the discord command. Will it be possible to generate `Writing`-Tokens for 3rd Party applications? If yes we have to make sure, that in this scenario no `Writing`-Token can be requested.

But for now I think its fine

Reply:
I think we already had the discussion with the outcome that only the Bot should be allowed to Write Data in the first steps. If we allow 3rd party apps just to read, theres only the `Data.Read`-Scope left. Thats why we can Hardcode it I think

Reply:
> that only the Bot should be allowed to Write Data in the first steps

oh yes, sorry I forgot

yes this makes sense 👍